### PR TITLE
Added workflow for GSSOC label

### PR DESCRIPTION
**Pull request by Abhijay007**
## What is the change?
Added workflow for GSSOC label

## Related issue?
issue: #237

## How was it tested?
tested on my local repo

## Checklist:
Before you create this PR, confirm all the requirements listed below by checking the checkboxes `[x]`:

- [x] Have you followed the [Contribution Guidelines](https://github.com/ALPHAVIO/BlogSite/blob/master/CONTRIBUTING.md) while contributing.
- [x] Have you checked there aren't other open [Pull Requests](https://github.com/ALPHAVIO/BlogSite/pulls) for the same update/change?
- [x] Your submission doesn't break any existing feature.
- [x] Have you tested the code before submission?

### CODE REVIEW

File: .github/workflows/autoLabel.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: Renato66/auto-label@v2
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          ignore-comments: true
+          labels-synonyms: '{"bug":["error","need fix","not working"],"enhancement":["upgrade"],"question":["help"]}'
+          labels-not-allowed: '["good first issue"]'
+          default-labels: '["GSSOC"]'
```
Comment (Pranav016) [inline .github/workflows/autoLabel.yaml:15]: Make default label `GSSOC21`

**Review by Pranav016 — CHANGES_REQUESTED** [review]
Does this action check if GSSOC is mentioned anywhere in the issue body ? It'll be great if you can refer the issue description again.

**Comment by Abhijay007** [comment]
> Does this action check if GSSOC is mentioned anywhere in the issue body ? It'll be great if you can refer the issue description again.

@Pranav016 sir I test with and without both GSSOC mention in the issue text.

***With***
with label

***Without***
without text

Sir see I raised this issue 3 minutes before that issue but as I did not mention GSSOC anywhere in the text the workflow didn't get triggered.

**Review by Pranav016 — APPROVED** [review]

**Review by Pranav016 — COMMENTED** [review]
@Abhijay007 Can you do one small things really quickly ? These labels should be added to any issue and 'any PR' that has "GSSOC","GSSOC21","gssoc","Gssoc","GSSoC" in their body. So just please make this small change and edit the action to add labels on pull request as well.
Since came to know late that labels need to be added on PRs as well so telling you know..

**Comment by Abhijay007** [comment]
> @Abhijay007 Can you do one small things really quickly ? These labels should be added to any issue and any PR that has "GSSOC","GSSOC21","gssoc","Gssoc","GSSoC" in their body. So just please make this small change and edit the action to add labels on pull request as well.
> Since came to know late that labels need to be added on PRs as well so telling you know..

ok sir I will do that, No problem!!

**Comment by Pranav016** [comment]
> > @Abhijay007 Can you do one small things really quickly ? These labels should be added to any issue and any PR that has "GSSOC","GSSOC21","gssoc","Gssoc","GSSoC" in their body. So just please make this small change and edit the action to add labels on pull request as well.
> > Since came to know late that labels need to be added on PRs as well so telling you know..
> 
> ok sir I will do that, No problem!!

Any progress ? The requested changes aren't that difficult. You just have to add to the `on` clause in the file. Let me know if you need any help

**Comment by Abhijay007** [comment]
> > > @Abhijay007 Can you do one small things really quickly ? These labels should be added to any issue and any PR that has "GSSOC","GSSOC21","gssoc","Gssoc","GSSoC" in their body. So just please make this small change and edit the action to add labels on pull request as well.
> > > Since came to know late that labels need to be added on PRs as well so telling you know..
> > 
> > 
> > ok sir I will do that, No problem!!
> 
> Any progress ? The requested changes aren't that difficult. You just have to add to the `on` clause in the file. Let me know if you need any help

sir, I tried to add the workflow for pull request by changing the `on` clause but when I run that workflow on my local repo that then that workflow crashes for every PR with or without GGSOC text in the body. then I try to make a separate workflow that is only for PR not issue included on the `on` clause but that also didn't work. I tried this process with almost 10 PRs in my local repo but not any PR get successful passed the labeler workflow checks. sir, I think that I need to find a new `uses` clause in order to set workflow for PRs.


thinks I tried:

pr-workflow1

with a separate workflow:
PR-workflow2


@Pranav016 sir, can you please cross-check it once on any other repo by changing the `on` clause for PR.

**Review by Pranav016 — COMMENTED** [review]
> > Any progress ? The requested changes aren't that difficult. You just have to add to the `on` clause in the file. Let me know if you need any help
> 
> sir, I tried to add the workflow for pull request by changing the `on` clause but when I run that workflow on my local repo that then that workflow crashes for every PR with or without GGSOC text in the body. then I try to make a separate workflow that is only for PR not issue included on the `on` clause but that also didn't work. I tried this process with almost 10 PRs in my local repo but not any PR get successful passed the labeler workflow checks. sir, I think that I need to find a new `uses` clause in order to set workflow for PRs.
> 
> thinks I tried:
> 
> pr-workflow1
> 
> with a separate workflow:
> PR-workflow2
> 
> @Pranav016 sir, can you please cross-check it once on any other repo by changing the `on` clause for PR.

Yes surely, I'll check whenever I get time.
One thing, was it working before you added the `pull_request` part ? I mean was it working fine for adding labels to issues ?

I'd suggest two things
1. Do like this 
image

1. Try removing the single quotation marks like this
image

I hope this resolves the issue and works fine

**Comment by Abhijay007** [comment]
> Yes surely, I'll check whenever I get time.
> One thing, was it working before you added the `pull_request` part ? I mean was it working fine for adding labels to issues ?

Yes, sir, it is working fine for the issue labeling.

**Comment by Abhijay007** [comment]
> I'd suggest two things
> 
> 1. Do like this
>    image
> 2. Try removing the single quotation marks like this
>    image
> 
> I hope this resolves the issue and works fine

ok sir I will try this .

**Comment by Abhijay007** [comment]
> I'd suggest two things
> 
> 1. Do like this
>    image
> 2. Try removing the single quotation marks like this
>    image
> 
> I hope this resolves the issue and works fine

No sir it didn't work, I don't know why but it runs fine with issue labeling but not for PR. Sir, can you please try this once for PRs labeling on any local repo whenever you get the time?

**Comment by Pranav016** [comment]
I'll take care of the PR part

**Comment by Abhijay007** [comment]
> I'll take care of the PR part

ok sir !!

**Comment by Pranav016** [comment]
I was testing this action and it auto added all the level labels as well 
image

Can you raise and issue for this and work on this bug again ?

**Comment by Abhijay007** [comment]
> I was testing this action and it auto added all the level labels as well which is not required.
> image
> 
> Can you raise and issue for this and work on this bug again ?

I don't know why label action for levels gets triggered 😅 as I didn't mention it anywhere in the workflow. 
when I tested it on some of my local repos nothing like that happened.  I think it happened due to other workflows that are defined under repo.
sir I will try to figure out why those workflows get triggered once I am done with changes I will open a PR for that. till then you can raise an issue for that if you want.

**Pull request merged**